### PR TITLE
Feature(Gradle): Upgrade gradle to version 6.8.1

### PR DESCRIPTION
**Pull request by DarkWeird**
# Description

* Upgrade gradle 6.8.1
* Upgrade findbugs -> spotbugs
* Update metrics version

# Testing
* Run/Build DestinationSol with modules

### CODE REVIEW

File: config/gradle/common.gradle [hunk 1]
```diff
@@ -111,22 +113,15 @@ tasks.withType(Checkstyle) {
...
     group = 'Reporting'
 }
+//
+//tasks.withType(SpotBugs) {
+//    group = 'Reporting'
+//    reports {
+//        xml.enabled = false
+//        html.enabled = true
+//    }
+//}
```
Comment (NicholasBatesNZ) [inline config/gradle/common.gradle:123]: Can this be removed?

File: desktop/build.gradle [hunk 2]
```diff
@@ -24,14 +24,13 @@ apply from: "../config/gradle/jre.gradle"
...
 
 dependencies {
-    compile project(":engine")
-    compile "com.badlogicgames.gdx:gdx-backend-lwjgl3:$gdxVersion"
-    compile "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-desktop"
-    compile "com.badlogicgames.gdx:gdx-box2d-platform:$gdxVersion:natives-desktop"
-    compile "com.badlogicgames.gdx:gdx-controllers-lwjgl3:$gdxVersion"
-
-    compile group: 'org.terasology.crashreporter', name: 'cr-destsol', version: '4.0.0'
-    compile group: 'org.slf4j', name: 'slf4j-log4j12', version: '1.7.25'
```
Comment (NicholasBatesNZ) [inline desktop/build.gradle:34]: Why do you remove this logger dependency?

File: engine/build.gradle [hunk 3]
```diff
@@ -33,52 +33,58 @@ buildscript {
...
+    api group: 'org.terasology.nui', name: 'nui', version: '2.0.0'
+    api group: 'org.terasology.nui', name: 'nui-libgdx', version: '2.0.0'
 
-    compile group: 'com.google.protobuf', name: 'protobuf-java', version: '3.4.0'
+    implementation group: 'com.google.protobuf', name: 'protobuf-java', version: '3.4.0'
 
-    compile "com.github.everit-org.json-schema:org.everit.json.schema:1.11.1"
-    compile "com.github.marschall:zipfilesystem-standalone:1.0.1"
-    compile 'dom4j:dom4j:1.6.1'
+    implementation "com.github.zafarkhaja:java-semver:0.9.0" // gestalt lost this...
```
Comment (NicholasBatesNZ) [inline engine/build.gradle:53]: Probably fine to leave this here but can it be added to gestalt instead or is it necessary here?

File: engine/build.gradle [hunk 4]
```diff
@@ -33,52 +33,58 @@ buildscript {
 }
 
 dependencies {
-    compile "com.badlogicgames.gdx:gdx:$gdxVersion"
-    compile "com.badlogicgames.gdx:gdx-box2d:$gdxVersion"
-    compile "com.badlogicgames.gdx:gdx-controllers:$gdxVersion"
+    api group: 'org.slf4j', name: 'slf4j-log4j12', version: '1.7.25'
+    api group: 'com.google.code.gson', name: 'gson', version: '2.6.2'
```
Comment (NicholasBatesNZ) [inline engine/build.gradle:37]: Where did these two come from?

**Review by NicholasBatesNZ — CHANGES_REQUESTED** [review]
Thanks so much for this, our gradle setup has been getting pretty dusty... :eyes: 
Unfortunately though the game won't run when the [android facade](https://github.com/MovingBlocks/DestSolAndroid/) is present. Presumably that needs to be updated too :/

**Review by BenjaminAmos — COMMENTED** [review]
I was just looking at this PR and noticed a couple of things, so I've left some comments. Modules appear to still work and the android facade works with this too after making a couple of small changes. I'll send a pull request to the DestSolAndroid repository with those changes.

File: engine/build.gradle [hunk 5]
```diff
@@ -33,52 +33,58 @@ buildscript {
...
-    // TODO: The extra "org" qualifier excludes test classes otherwise sucked up in Jenkins, causing issues. Redo later
-    File dirToReflect = new File(sourceSets.main.output.classesDir, "org")
-    inputs.files dirToReflect
-    outputs.file file(sourceSets.main.output.classesDir.toString() + "/reflections.cache")
+    inputs.files sourceSets.main.output.classesDirs,
+            // getClassesDir from all sourceSets (for any jvm (seems) language)
+            configurations."${sourceSets.main.runtimeClasspathConfigurationName}"
+
+    outputs.upToDateWhen {classes.state.upToDate}
+    outputs.file("$buildDir/resources/main/reflections.cache")
```
Comment (BenjaminAmos) [inline engine/build.gradle:79]: Where is `$buildDir/resources/main/reflections.cache` created? It seems that the reflections are saved in a different location. Did you mean to use `$buildDir/classes/reflections.cache` instead?
```suggestion
    outputs.file("$buildDir/classes/reflections.cache")
```

File: templates/build.gradle [hunk 6]
```diff
@@ -190,24 +187,26 @@ task createSkeleton() {
...
-    // TODO: The extra "org" qualifier excludes test classes otherwise sucked up in Jenkins, causing issues. Redo later
-    File dirToReflect = new File(sourceSets.main.output.classesDir, "org")
-    inputs.files dirToReflect
-    outputs.file file(sourceSets.main.output.classesDir.toString() + "/reflections.cache")
+    inputs.files sourceSets.main.output.classesDirs,
+            // getClassesDir from all sourceSets (for any jvm (seems) language)
+            configurations."${sourceSets.main.runtimeClasspathConfigurationName}"
+
+    outputs.upToDateWhen {classes.state.upToDate}
+    outputs.file("$buildDir/resources/main/reflections.cache")
```
Comment (BenjaminAmos) [inline templates/build.gradle:195]: Where is `$buildDir/resources/main/reflections.cache` created? It seems that the reflections are saved in a different location. Did you mean to use `$buildDir/classes/reflections.cache` instead?
```suggestion
    outputs.file("$buildDir/classes/reflections.cache")
```

File: desktop/build.gradle (hunk 2 above)
Comment (BenjaminAmos) [inline desktop/build.gradle:34]: This shouldn't have been moved into the engine `build.gradle`. This specifies the log4j back-end for the slf4j library, which is only used with the desktop facade. Different facades can use different slf4j logging back-ends. See http://www.slf4j.org/manual.html#swapping for more information.

File: engine/build.gradle [hunk 7]
```diff
@@ -33,52 +33,58 @@ buildscript {
 }
 
 dependencies {
-    compile "com.badlogicgames.gdx:gdx:$gdxVersion"
-    compile "com.badlogicgames.gdx:gdx-box2d:$gdxVersion"
-    compile "com.badlogicgames.gdx:gdx-controllers:$gdxVersion"
+    api group: 'org.slf4j', name: 'slf4j-log4j12', version: '1.7.25'
```
Comment (BenjaminAmos) [inline engine/build.gradle:36]: The engine should not depend on a specific logging back-end of slf4j. Log4j caused errors at runtime with the android facade when I tried to run this code. Only the desktop facade uses log4j for logging so far. The engine should depend on `slf4j-api` instead. See http://www.slf4j.org/manual.html#swapping for more information.
```suggestion
    api group: 'org.slf4j', name: 'slf4j-api', version: '1.7.25'
```

File: engine/build.gradle (hunk 3 above)
Comment (BenjaminAmos) [inline engine/build.gradle:53]: This is different to the JSemver version that gestalt uses. Gestalt uses JSemver `0.10.0` from the `heisluft.de` repository.
```suggestion
    implementation "com.github.zafarkhaja:java-semver:0.10.0" // gestalt lost this...
```
https://github.com/MovingBlocks/gestalt/blob/fc5e69a197eb37456284fb80ea1dd7d53b3d410c/gestalt-module/build.gradle#L43

**Review by BenjaminAmos — COMMENTED** [review]
For some reason my previous comments didn't save, so hopefully they will show this time.

**Comment by BenjaminAmos** [comment]
The android changes to go with this pull request are at https://github.com/MovingBlocks/DestSolAndroid/pull/17.

File: desktop/build.gradle (hunk 2 above)
Comment (DarkWeird) [reply desktop/build.gradle:34]: It provides by engine.

File: engine/build.gradle (hunk 3 above)
Comment (DarkWeird) [reply engine/build.gradle:53]: This some kind magic T_T.
faster left it there.

File: engine/build.gradle (hunk 4 above)
Comment (DarkWeird) [reply engine/build.gradle:37]: slf4j from desktop module. used in `engine`.
gson required for UIFormat and etc. idk where it was.

**Review by NicholasBatesNZ — APPROVED** [review]
Thanks

**Pull request merged**